src/content-items/utils.ts

```diff
@@ -52,3 +55,19 @@ export async function getRawMetadata(
     metadata,
   }
 }
+
+/**
+ * Gets shared information about pod or file
+ *
+ * @param bee Bee instance
+ * @param reference reference to shared information
+ */
+export async function getSharedInfo(bee: Bee, reference: string): Promise<PodShareInfo | FileShareInfo> {
+  const data = (await bee.downloadData(reference)).json()
+
+  if (!(isPodShareInfo(data) || isFileShareInfo(data))) {
```

Comment:
because of this you could split this `getSharedInfo` that would assert file and podsharing separately.
I'm telling this because after calling `getSharedInfo` you need to assert again for one of both